src/etc/inc/notices.inc

```diff
@@ -410,10 +410,64 @@ function send_smtp_message($message, $subject = "(no subject)", $force = false)
 	log_error(sprintf(gettext("Message sent to %s OK"), $to));
 	return;
 }
+/****f* notices/notify_via_telegram
+ * NAME
+ *   notify_via_telegram
+ * INPUTS
+ *	 notification string to send to Telegram via API
+ * RESULT
+ *   returns NULL if message was sent
+ ******/
+
+function notify_via_telegram($message, $force = false){
+	global $config;
+	if (isset($config['notifications']['telegram']['disable']) && !$force) {
+		return;
+	}
+	if (!$config['notifications']['telegram']['api']) {
+		return;
+	}
+	if (!$config['notifications']['telegram']['chatid']) {
+		return;
+	}
+	$url = "https://api.telegram.org/bot{$config['notifications']['telegram']['api']}/sendMessage?";
+	$data = array(
+		"chat_id" => $config['notifications']['telegram']['chatid'],
+		"text" => "{$config['system']['hostname']}.{$config['system']['domain']}\n{$message}"
+	);
+	$result = json_decode(curl_post_telegram($url . http_build_query($data)), true);
+	if ($result['ok']) {
```

Comment:
This should test if `$result` is an array before treating it as one. If the `json_decode()` fails, `$result` would be `null` and this would result in a PHP error.